git_machete/cmd.py

```diff
@@ -1494,16 +1526,16 @@ def is_merged_to(b, target):
         return is_ancestor(b, target)
 
 
-def is_merged_to_upstream(b):
+def is_merged_to_upstream(b: str) -> bool:
     if b not in up_branch:
         return False
     return is_merged_to(b, up_branch[b])
 
 
-def infer_upstream(b, condition=lambda u: True, reject_reason_message=""):
+def infer_upstream(b: str, condition: Callable[..., bool] = lambda u: True, reject_reason_message: str = "") -> Optional[str]:
```

Comment:
Hmm why `...` ? is it some kind of varargs?

Reply:
Shouldn't the Callable's argument type be just `str` here?

Reply:
Oh ok `...` is just _anything_ AFAICS

Reply:
Yup (it is called `Ellipsis` IIRC). I tried to be more precise here but I am afraid that it's not possible (yet)

Reply:
Ok.... so what's the error if that's replaced with `Callable[str, bool]`? Or maybe just try `Callable[Any, bool]`?